File: backend/src/auth/dto/dto/update-user.dto.ts
import { IsOptional, IsString } from 'class-validator';

export class UpdateUserDTO {
  @IsOptional()
  @IsString()
  name: string;
  @IsOptional()
  @IsString()
  surname: string;
  @IsOptional()
  @IsString()
  phone: string;
}

File: fontend/lib/domain/dtos/update_user_dto.dart
import 'dart:io';

class UpdateUserDTO {
  String? name;
  String? surname;
  String? phone;
  File? file;

  UpdateUserDTO({
    this.name,
    this.surname,
    this.phone,
    this.file,
  });

  Map<String, dynamic> topMap() {
    return {
      'name': name,
      'surname': surname,
      'phone': phone,
    };
  }
}
